pushmsixscript/src/pushmsixscript/microsoft_store.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import time
+
+import requests
+from scriptworker_client.exceptions import TaskVerificationError
+
+from pushmsixscript import task
+
+log = logging.getLogger(__name__)
+
+# When committing a new submission, poll for completion, with
+# this many attempts, waiting this long between attempts.
+COMMIT_POLL_MAX_ATTEMPTS = 10
+COMMIT_POLL_WAIT_SECONDS = 30
+
+
+def push(config, msix_file_path, channel):
+    """Publishes a msix onto a given channel.
+
+    This function performs all the network actions to ensure `msix_file_path` is published on
+    `channel`. If `channel` is not whitelisted to contact the Microsoft Store, then it early
+    returns (this allows staging tasks to not contact the Microsoft Store at all).
+
+    Args:
+        config (config): the scriptworker configuration.
+        msix_file_path (str): The full path to the msix file to upload.
+        channel (str): The Microsoft Store channel.
+    """
+    if not task.is_allowed_to_push_to_microsoft_store(config, channel=channel):
+        log.warning("Not allowed to push to Microsoft Store. Skipping push...")
+        # We don't raise an error because we still want green tasks on dev instances
+        return
+
+    access_token = _store_session(config)
+    if access_token:
+        _push_to_store(config, channel, msix_file_path, access_token)
+    else:
+        raise TaskVerificationError("unable to push: missing access token")
+
+
+def _store_url(config, tail):
+    return config["store_url"] + tail
+
+
+def _log_response(response):
+    log.info(f"response code: {response.status_code}")
+    log.info(f"response headers: {response.headers}")
+    log.info(f"response body: {response.text}")
+
+
+def _store_session(config):
+    tenant_id = config["tenant_id"]
+    client_id = config["client_id"]
+    client_secret = config["client_secret"]
+    login_url = config["login_url"]
+    token_resource = config["token_resource"]
+    url = f"{login_url}/{tenant_id}/oauth2/token"
+    body = f"grant_type=client_credentials&client_id={client_id}&client_secret={client_secret}&resource={token_resource}"
+    headers = {"Content-Type": "application/x-www-form-urlencoded; charset=utf-8"}
+    response = requests.post(url, body, headers=headers, timeout=config["request_timeout_seconds"])
+    _log_response(response)
```

Comment:
Have we verified that no sensitive information gets logged? Otherwise we may want to modify what we log / scrub the response.

Reply:
I reviewed the logging and removed this call in particular.